lsp-mode.el

```diff
@@ -5899,7 +5915,42 @@ Request codeAction/resolve for more info if server supports."
 
   (cond
    ((stringp command?) (lsp--execute-command action))
-   ((lsp-command? command?) (lsp--execute-command command?))))
+   ((lsp-command? command?) (progn
+                              (lsp--fix-code-action-booleans command?)
+                              (lsp--execute-command command?)))))
+
+(lsp-defun lsp--fix-code-action-booleans ((&Command :arguments?))
+  "Patch incorrect boolean argument values in the provided `CodeAction' command
+in place, based on the BOOLEAN-ACTION-ARGUMENTS LSP client
+parameter. lsp-mode parses JSON false values `nil' values, which are then
+encoded back to JSON as null values. This may not match the language server's
+schema, and these values thus need to be changed patched back to false values
+for the affected requests to work correctly."
+  (let ((boolean-action-arguments (->> (lsp-workspaces)
+                                       (cl-first)
+                                       (or lsp--cur-workspace)
+                                       (lsp--workspace-client)
+                                       (lsp--client-boolean-action-arguments))))
+    (seq-doseq (args arguments?)
+      (seq-doseq (path boolean-action-arguments)
+        (print (if (listp path) path (list path)))
+        (lsp--fix-nested-boolean args (if (listp path) path (list path)))))))
+
+(defun lsp--fix-nested-boolean (structure path)
+  "Traverse STRUCTURE using the paths from the PATH list, changing the value to
+`:json-false' if it was `nil'. PATH should be a list containing
+one or more symbols, and STRUCTURE should be compatible with
+`lsp-member?', `lsp-get', and `lsp-put'."
+  (let ((key (car path))
+        (rest (cdr path)))
+    (if (null rest)
+        ;; `lsp-put' returns `nil' both when the key doesn't exist and when the
+        ;; value is `nil', so we need to explicitly check its presence here
+        (when (and (lsp-member? structure key) (not (lsp-get structure key)))
+          (lsp-put structure key :json-false)))
+    ;; If `key' does not exist, then we'll silently ignore it
+    (when-let ((child (lsp-get structure key)))
+      (lsp--fix-nested-boolean child rest))))
```

Comment:
I'm not very familiar with elisp and its data structures, so maybe there's an existing/better combinator for doing deep queries/updates in nested hash tables.